### ISSUE TITLE
Crash reading CODE128 barcode on iPhone5

### ISSUE THREAD

**Issue by mrbelk** [Redth/ZXing.Net.Mobile#171]
I am trying to build a Xamarin Forms app (iOS and Android) that incorporates your barcode scanning library using aritchie's Xamarin Forms port. There is a particular code128 barcode that I cannot scan using an iPhone 5 (iOS 8.3) that scans perfectly fine on a Nexus 5 running 5.1.1. My small test app targets Android API 19 and iOS 8. I have restricted ZXing from considering any symbologies other than code 39 and code 128.

What happens on iOS is that the app crashes with an unhandled NullReferenceException. I have wrapped the actual reading code in a try...catch but can't catch the error.

I don't think I can generate a copy of the document that is readable, but the value in the barcode is:

CX-S749+D3035-HARRI+I7149241

Thanks.

**Comment by mrbelk** [comment]
For some context, here is a stack trace from AppDomain.CurrentDomain.UnhandledException

```
System.NullReferenceException: Object reference not set to an instance of an object

[0:] 
2015-06-16 09:15:26.291 BarcodeTestiOS[385:197564] System.UnhandledExceptionEventArgs
[0:] System.UnhandledExceptionEventArgs
2015-06-16 09:15:26.353 BarcodeTestiOS[385:197564] Unhandled managed exception:
Object reference not set to an instance of an object (System.NullReferenceException)
  at ZXing.Mobile.ZXingScannerView.ResizePreview (UIInterfaceOrientation orientation) [0x000c6] in <filename unknown>:0 
  at ZXing.Mobile.ZXingScannerView.DidRotate (UIInterfaceOrientation orientation) [0x00000] in <filename unknown>:0 
  at ZXing.Mobile.ZXingScannerViewController.DidRotate (UIInterfaceOrientation fromInterfaceOrientation) [0x00017] in <filename unknown>:0 
  at (wrapper managed-to-native) ObjCRuntime.Messaging:void_objc_msgSendSuper_bool_IntPtr (intptr,intptr,bool,intptr)
  at UIKit.UIViewController.DismissViewController (Boolean animated, System.Action completionHandler) [0x00057] in /Users/builder/data/lanes/1503/e6ebd18b/source/maccore/src/build/ios/native/UIKit/UIViewController.g.cs:330 
  at ZXing.Mobile.MobileBarcodeScanner+<Scan>c__AnonStorey0+<Scan>c__AnonStorey1+<Scan>c__AnonStorey2.<>m__0 () [0x00015] in <filename unkn
own>:0 
  at Foundation.NSActionDispatcher.Apply () [0x00000] in /Users/builder/data/lanes/1503/e6ebd18b/source/maccore/src/Foundation/NSAction.cs:56 
  at (wrapper managed-to-native) UIKit.UIApplication:UIApplicationMain (int,string[],intptr,intptr)
  at UIKit.UIApplication.Main (System.String[] args, IntPtr principal, IntPtr delegate) [0x00005] in /Users/builder/data/lanes/1503/e6ebd18b/source/maccore/src/UIKit/UIApplication.cs:63 
  at UIKit.UIApplication.Main (System.String[] args, System.String principalClassName, System.String delegateClassName) [0x0001c] in /Users/builder/data/lanes/1503/e6ebd18b/source/maccore/src/UIKit/UIApplication.cs:46 
  at BarcodeTest.iOS.Application.Main (System.String[] args) [0x00001] in c:\Users\mbelk.BIZSPEED\My Source Code\Allmed Express\Xamarin\BarcodeTest\BarcodeTest\BarcodeTest.iOS\Main.cs:17 
2015-06-16 09:15:26.356 BarcodeTestiOS[385:197564] critical: Stacktrace:

2015-06-16 09:15:26.357 BarcodeTestiOS[385:197564] critical: 
Native stacktrace:
2015-06-16 09:15:26.614 BarcodeTestiOS[385:197564] critical:    0   BarcodeTestiOS                      0x00e150b7 mono_handle_native_sigsegv + 238
2015-06-16 09:15:26.615 BarcodeTestiOS[385:197564] critical:    1   BarcodeTestiOS                      0x00e18f85 sigabrt_signal_handler + 104
2015-06-16 09:15:26.616 BarcodeTestiOS[385:197564] critical:    2   libsystem_platform.dylib            0x3194e87b _sigtramp + 42
2015-06-16 09:15:26.616 BarcodeTestiOS[385:197564] critical:    3   libsystem_pthread.dylib             0x31952cc7 pthread_kill + 62
2015-06-16 09:15:26.617 BarcodeTestiOS[385:197564] critical:    4   libsystem_c.dylib                   0x3186f909 abort + 76
2015-06-16 09:15:26.617 BarcodeTestiOS[385:197564] critical:    5   BarcodeTestiOS                      0x00eb6f90 print_exception + 0
2015-06-16 09:15:26.618 BarcodeTestiOS[385:197564] critical:    6   BarcodeTestiOS                      0x00e15623 mono_invoke_unhandled_exception_hook + 66
2015-06-16 09:15:26.618 BarcodeTestiOS[385:197564] critical:    7   BarcodeTestiOS                      0x00e1faaf mono_thread_abort_dummy + 62
2015-06-16 09:15:26.619 BarcodeTestiOS[385:197564] critical:    8   Bar
codeTestiOS                      0x00e14e45 mono_handle_exception_internal + 3784
2015-06-16 09:15:26.619 BarcodeTestiOS[385:197564] critical:    9   BarcodeTestiOS                      0x00e13f77 mono_handle_exception + 30
2015-06-16 09:15:26.620 BarcodeTestiOS[385:197564] critical:    10  BarcodeTestiOS                      0x00e0ebdd handle_signal_exception + 48
2015-06-16 09:15:26.620 BarcodeTestiOS[385:197564] critical:    11  ???                                 0x00000000 0x0 + 0
2015-06-16 09:15:26.621 BarcodeTestiOS[385:197564] critical:    12  BarcodeTestiOS                      0x00d9c530 ZXing_Mobile_ZXingScannerView_DidRotate_UIKit_UIInterfaceOrientation + 116
2015-06-16 09:15:26.621 BarcodeTestiOS[385:197564] critical:    13  BarcodeTestiOS                      0x00da2868 ZXing_Mobile_ZXingScannerViewController_DidRotate_UIKit_UIInterfaceOrientation + 212
2015-06-16 09:15:26.622 BarcodeTestiOS[385:197564] critical:    14  BarcodeTestiOS                      0x002a41a0 wrapper_runtime_invoke_object_runtime_invoke_dynamic_intptr_intptr_intptr_intptr + 224
2015-06-16 09:15:26.622 BarcodeTestiOS[385:197564] critical:    15  BarcodeTestiOS                      0x00e1f43b mono_jit_runtime_invoke + 1190
2015-06-16 09:15:26.623 BarcodeTestiOS[385:197564] critical:    16  BarcodeTestiOS                      0x00e5e121 mono_runtime_invoke + 88
2015-06-16 09:15:26.624 BarcodeTestiOS[385:197564] critical:    17  BarcodeTestiOS                      0x00ddfb57 native_to_managed_trampoline_16 + 186
2015-06-16 09:15:26.624 BarcodeTestiOS[385:197564] critical:    18  BarcodeTestiOS                      0x00dedf71 -[ZXing_Mobile_ZXingScannerViewController didRotateFromInterfaceOrientation:] + 80
2015-06-16 09:15:26.625 BarcodeTestiOS[385:197564] critical:    19  UIKit                               0x26708af1 <redacted> + 168
2015-06-16 09:15:26.625 BarcodeTestiOS[385:197564] critical:    20 
 UIKit                               0x267323a9 <redacted> + 116
2015-06-16 09:15:26.626 BarcodeTestiOS[385:197564] critical:    21  UIKit                               0x26731381 <redacted> + 120
2015-06-16 09:15:26.626 BarcodeTestiOS[385:197564] critical:    22  UIKit                               0x26ae482b <redacted> + 46
2015-06-16 09:15:26.627 BarcodeTestiOS[385:197564] critical:    23  UIKit                               0x26ae464d <redacted> + 424
2015-06-16 09:15:26.627 BarcodeTestiOS[385:197564] critical:    24  UIKit                               0x268b2a6d <redacted> + 760
2015-06-16 09:15:26.628 BarcodeTestiOS[385:197564] critical:    25  UIKit                               0x268b3a4f <redacted> + 1558
2015-06-16 09:15:26.628 BarcodeTestiOS[385:197564] critical:    26  UIKit                               0x268b412b <redacted> + 538
2015-06-16 09:15:26.629 BarcodeTestiOS[385:197564] critical:    27  UIKit                               0x26697033 <redacted> + 98
2015-06-16 09:15:26.629 BarcodeTestiOS[385:197564] critical:    28  UIKit                               0x26696fcb <redacted> + 46
2015-06-16 09:15:26.630 BarcodeTestiOS[385:197564] critical:    29  UIKit                               0x26696e9b <redacted> + 118
2015-06-16 09:15:26.630 BarcodeTestiOS[385:197564] critical:    30  UIKit                               0x26696e1b <redacted> + 42
2015-06-16 09:15:26.631 BarcodeTestiOS[385:197564] critical:    31  UIKit                               0x26c7d8b7 <redacted> + 158
2015-06-16 09:15:26.631 BarcodeTestiOS[385:197564] critical:    32  UIKit                               0x26c7d79f <redacted> + 626
2015-06-16 09:15:26.632 BarcodeTestiOS[385:197564] critical:    33  UIKit                               0x268f668f <red
acted> + 366
2015-06-16 09:15:26.632 BarcodeTestiOS[385:197564] critical:    34  UIKit                               0x269111ed <redacted> + 504
2015-06-16 09:15:26.633 BarcodeTestiOS[385:197564] critical:    35  UIKit                               0x267366bd <redacted> + 724
2015-06-16 09:15:26.633 BarcodeTestiOS[385:197564] critical:    36  UIKit                               0x266e3b9b <redacted> + 906
2015-06-16 09:15:26.634 BarcodeTestiOS[385:197564] critical:    37  UIKit                               0x266e3b9b <redacted> + 906
2015-06-16 09:15:26.634 BarcodeTestiOS[385:197564] critical:    38  UIKit                               0x266e37b9 <redacted> + 224
2015-06-16 09:15:26.635 BarcodeTestiOS[385:197564] critical:    39  BarcodeTestiOS                      0x004e27a4 wrapper_managed_to_native_ObjCRuntime_Messaging_void_objc_msgSendSuper_bool_IntPtr_intptr_intptr_bool_intptr + 136
2015-06-16 09:15:26.635 BarcodeTestiOS[385:197564] critical:    40  BarcodeTestiOS                      0x0048c760 UIKit_UIViewController_DismissViewController_bool_System_Action + 336
2015-06-16 09:15:26.636 BarcodeTestiOS[385:197564] critical:    41  BarcodeTestiOS                      0x00db67b4 ZXing_Mobile_MobileBarcodeScanner__Scanc__AnonStorey0__Scanc__AnonStorey1__Scanc__AnonStorey2__m__0 + 412
2015-06-16 09:15:26.637 BarcodeTestiOS[385:197564] critical:    42  BarcodeTestiOS                      0x0044ba40 Foundation_NSActionDispatcher_Apply + 44
2015-06-16 09:15:26.637 BarcodeTestiOS[385:197564] critical:    43  BarcodeTestiOS                      0x002a41a0 wrapper_runtime_invoke_object_runtime_invoke_dynamic_intptr_intptr_intptr_intptr + 224
2015-06-16 09:15:26.638 BarcodeTestiOS[385:197564] critical:    44  BarcodeTestiOS                      0x00e1f43b mono_jit_runtime_invoke + 1190
2015-06-16 09:15:26.638 BarcodeTestiOS[385:197564] critical:    45  BarcodeTestiOS                      0x00e5e121 mono_runtim
e_invoke + 88
2015-06-16 09:15:26.639 BarcodeTestiOS[385:197564] critical:    46  BarcodeTestiOS                      0x00ddf0d1 native_to_managed_trampoline_5 + 128
2015-06-16 09:15:26.639 BarcodeTestiOS[385:197564] critical:    47  BarcodeTestiOS                      0x00de9e2d -[__MonoMac_NSActionDispatcher xamarinApplySelector] + 52
2015-06-16 09:15:26.640 BarcodeTestiOS[385:197564] critical:    48  Foundation                          0x23d0a5cf <redacted> + 386
2015-06-16 09:15:26.640 BarcodeTestiOS[385:197564] critical:    49  CoreFoundation                      0x22f91faf <redacted> + 14
2015-06-16 09:15:26.641 BarcodeTestiOS[385:197564] critical:    50  CoreFoundation                      0x22f913bf <redacted> + 218
2015-06-16 09:15:26.641 BarcodeTestiOS[385:197564] critical:    51  CoreFoundation                      0x22f8fa25 <redacted> + 772
2015-06-16 09:15:26.642 BarcodeTestiOS[385:197564] critical:    52  CoreFoundation                      0x22edc201 CFRunLoopRunSpecific + 476
2015-06-16 09:15:26.642 Barcod
eTestiOS[385:197564] critical:  53  CoreFoundation                      0x22edc013 CFRunLoopRunInMode + 106
2015-06-16 09:15:26.643 BarcodeTestiOS[385:197564] critical:    54  GraphicsServices                    0x2a7bb201 GSEventRunModal + 136
2015-06-16 09:15:26.643 BarcodeTestiOS[385:197564] critical:    55  UIKit                               0x26680a59 UIApplicationMain + 1440
2015-06-16 09:15:26.644 BarcodeTestiOS[385:197564] critical:    56  BarcodeTestiOS                      0x004e8384 wrapper_managed_to_native_UIKit_UIApplication_UIApplicationMain_int_string___intptr_intptr + 272
2015-06-16 09:15:26.644 BarcodeTestiOS[385:197564] critical:    57  BarcodeTestiOS                      0x0046a568 UIKit_UIApplication_Main_string___intptr_intptr + 52
2015-06-16 09:15:26.645 BarcodeTestiOS[385:197564] critical:    58  BarcodeTestiOS                      0x0046a528 UIKit_UIApplication_Main_string___string_string + 204
2015-06-16 09:15:26.645 BarcodeTestiOS[385:197564] critical:    59  BarcodeTestiOS                    
  0x000c2180 BarcodeTest_iOS_Application_Main_string__ + 140
2015-06-16 09:15:26.646 BarcodeTestiOS[385:197564] critical:    60  BarcodeTestiOS                      0x002a41a0 wrapper_runtime_invoke_object_runtime_invoke_dynamic_intptr_intptr_intptr_intptr + 224
2015-06-16 09:15:26.647 BarcodeTestiOS[385:197564] critical:    61  BarcodeTestiOS                      0x00e1f43b mono_jit_runtime_invoke + 1190
2015-06-16 09:15:26.647 BarcodeTestiOS[385:197564] critical:    62  BarcodeTestiOS                      0x00e5e121 mono_runtime_invoke + 88
2015-06-16 09:15:26.648 BarcodeTestiOS[385:197564] critical:    63  BarcodeTestiOS                      0x00e615ef mono_runtime_exec_main + 282
2015-06-16 09:15:26.648 BarcodeTestiOS[385:197564] critical:    64  BarcodeTestiOS                      0x00e61431 mono_runtime_run_main + 476
2015-06-16 09:15:26.649 BarcodeTestiOS[385:197564] critical:    65  BarcodeTestiOS                      0x00e0e131 mono_jit_exec + 48
2015-06-16 09:15:26.649 BarcodeTestiOS[385:197564] critical:    66  BarcodeTestiOS                      0x00eb473c xamarin_main + 2184
2015-06-16 09:15:26.650 BarcodeTestiOS[385:197564] critical:    67  BarcodeTestiOS                      0x00df1039 main + 112
2015-06-16 09:15:26.650 BarcodeTestiOS[385:197564] critical:    68  libdyld.dylib                       0x31809aaf <redacted> + 2
2015-06-16 09:15:26.651 BarcodeTestiOS[385:197564] critical: 
=================================================================
Got a SIGABRT while executing native code. This usually indicates
a fatal error in the mono runt
ime or one of the native libraries 
used by your application.
=================================================================
```

**Comment by Redth** [comment]
This actually looks like #162 and not related to the barcode itself.  This particular crash is fixed in v1.5.0 which will be released shortly.
